logstash-core/src/main/java/org/logstash/ext/JrubyWrappedSynchronousQueueExt.java

```diff
@@ -0,0 +1,49 @@
+package org.logstash.ext;
+
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyNumeric;
+import org.jruby.RubyObject;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+@JRubyClass(name = "WrappedSynchronousQueue")
+public class JrubyWrappedSynchronousQueueExt extends RubyObject {
+
+    private BlockingQueue<JrubyEventExtLibrary.RubyEvent> queue;
+
+    public JrubyWrappedSynchronousQueueExt(final Ruby runtime, final RubyClass metaClass) {
+        super(runtime, metaClass);
+    }
+
+    @JRubyMethod(name = "initialize")
+    @SuppressWarnings("unchecked")
+    public void rubyInitialize(final ThreadContext context, IRubyObject size) {
+        int typedSize = ((RubyNumeric)size).getIntValue();
+        this.queue = new ArrayBlockingQueue<>(typedSize);
+    }
+
+    @JRubyMethod(name = "write_client")
+    public IRubyObject getWriteClient(final ThreadContext context) {
+        return JrubyMemoryWriteClientExt.create(queue);
+    }
+
+    @JRubyMethod(name = "read_client")
+    public IRubyObject getReadClient(final ThreadContext context) {
+        // batch size and timeout are currently hard-coded to 125 and 50ms as values observed
+        // to be reasonable tradeoffs between latency and throughput per PR #8707
+        return JrubyMemoryReadClientExt.create(queue, 125, 50);
```

Comment:
@danhermann ^^ here

Reply:
@danhermann yikes missed that one, we need to fix this by passing down the actual setting values here I'm afraid.

Reply:
@jakelandis, while the `MemoryReadClient` exposes a configurable batch size and read timeout, it was never called in 6.0 with anything other than the default settings of 125 and 50, respectively (see link below). The port to Java retains that same behavior.

https://github.com/elastic/logstash/blob/6.0/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb#L30

Reply:
@danhermann there was a `setter` for these settings that was called when setting the worker threads (you are right the constructor always used the defaults and was super redundant). I'm afraid we had some merge conflict at some point and lost the change that fixed the invocation of the constructor and simply kept the removal of that setter.
=> the problem isn't really in this PR but happened earlier but we should still fix this :)

Reply:
@jakelandis actually nevermind this was my mistake, we still have https://github.com/elastic/logstash/blob/master/logstash-core/src/main/java/org/logstash/execution/QueueReadClientBase.java#L70 and that is called from `java_pipeline.rb` so we're good here imo unless it's broken for you in practice?

Reply:
@danhermann @original-brownbear I just saw this while looking at something unrelated, and it seemed off, I didn't run any tests. Apologies if I misunderstood the change.